config.m4

```diff
@@ -329,9 +332,21 @@ if test "$MONGODB" != "no"; then
 
     AC_CHECK_TYPE([socklen_t], [AC_SUBST(MONGOC_HAVE_SOCKLEN, 1)], [AC_SUBST(MONGOC_HAVE_SOCKLEN, 0)], [#include <sys/socket.h>])
 
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION_SNAPPY, 0)
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION_ZLIB, 0)
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION, 0)
+    with_snappy=auto
+    with_zlib=auto
```

Comment:
Note: I manually tested this PR by changing this line to `with_zlib=bundled` since my system has zlib installed and `configure` picks that up first. Examining the `configure` output, I then checked for `whether to enable bundled zlib... 
yes` and verified that `MONGODB_SHARED_LIBADD` did not include `-lz`.

Since we don't report compression engines anywhere, runtime testing entailed tweaking `tests/connect/compression_error-001.phpt` to check that `zlib` was still accepted when building the driver with bundled zlib (it was).

Reply:
I think we should report compression engines then?

Reply:
Opened [PHPC-1097](https://jira.mongodb.org/browse/PHPC-1097).